packages/dds/tree/src/util/utils.ts

```diff
@@ -98,6 +98,13 @@ export function hasSome<T>(array: readonly T[]): array is [T, ...T[]] {
 	return array.length > 0;
 }
 
+/**
+ * Returns true if and only if the given iterable has at least one element.
+ */
+export function iterableHasSome<T>(iterable: Iterable<T>): boolean {
```

Comment:
It'd be slick if this were just another overload of `hasSome` (put it below the `array: readonly T[]` variant so it has lowest precedence). This implementation should work for all of the `hasSome` overloads. I know you are wary of overloads 😅, but in this case it seems like it'd be a net good.